modules/giapi/src/main/scala/giapi/client/GiapiStatusDb.scala

```diff
@@ -0,0 +1,119 @@
+import fs2.Stream
+
+/////////////////////////////////////////////////////////////////
+// Links status streaming with the giapi db
+/////////////////////////////////////////////////////////////////
+trait GiapiStatusDb[F[_]] {
+  def value(i: String): F[Option[StatusValue]]
+  def close: F[Unit]
+}
+
+object GiapiStatusDb {
+  private def dbUpdate[F[_]: Applicative, A](db:   GiapiDb[F],
+                                             name: String,
+                                             a:    A): F[Unit] =
+    a match {
+      case a: Int =>
+        db.update[Int](name, a)
+      case a: String =>
+        db.update[String](name, a)
+      case a: Float =>
+        db.update[Float](name, a)
+      case a: Double =>
+        db.update[Double](name, a)
+      case _ =>
+        Applicative[F].unit
+    }
+
+  def streamItemsToDb[F[_]: ConcurrentEffect](agg:   StatusHandlerAggregate,
+                                              db:    GiapiDb[F],
+                                              items: List[String]): F[Unit] = {
+    def statusHandler(q: Queue[F, (String, _)]) = new StatusHandler {
+
+      override def update[B](item: StatusItem[B]): Unit =
+        // Check the item name and enqueue it
+        if (items.contains(item.getName)) {
+          Option(item.getValue).foreach { a =>
+            q.enqueue1((item.getName, a)).toIO.unsafeRunAsync(_ => ())
+          }
+        }
+
+      override def getName: String = "Giapi staus db"
+    }
+
+    // A trivial resource that binds and unbinds a status handler.
+    def bind(q: Queue[F, (String, _)]): Resource[F, StatusHandler] =
+      Resource.make(
+        Effect[F].delay {
+          val sh = statusHandler(q)
+          agg.bindStatusHandler(sh)
+          sh
+        }
+      )(sh => Effect[F].delay(agg.unbindStatusHandler(sh)))
+
+    // Create a queue and put updates to forward them to the db
+    val s = for {
+      q <- Stream.eval(Queue.unbounded[F, (String, _)])
+      r <- Stream.resource(bind(q))
+      _ <- q.dequeue.map { case (n, v) => dbUpdate(db, n, v) }
+    } yield r
+    s.compile.drain
+  }
+
+  /**
+    * Creates a new status db in simulation
+    */
+  def simulatedDb[F[_]: Applicative]: GiapiStatusDb[F] =
+    new GiapiStatusDb[F] {
+      def value(i: String): F[Option[StatusValue]] =
+        Applicative[F].pure(none)
+
+      def close: F[Unit] = Applicative[F].pure(())
+    }
+
+  /**
+    * Creates a new status db that listens for status items as they are produced
+    *
+    * @param url Url of the giapi server
+    * @param items List of items to monitor
+    */
+  def newStatusDb[F[_]: ConcurrentEffect](
+    url:   String,
+    items: List[String]
+  ): F[GiapiStatusDb[F]] =
+    for {
+      c  <- Sync[F].delay(new ActiveMQJmsProvider(url)) // Build the connection
+      ss <- Giapi.statusStreamer[F](c) // giapi artifacts
+      db <- GiapiDb.newDb
+      f  <- streamItemsToDb[F](ss.aggregate, db, items).start // run in the background
```

Comment:
I want this to run in the background. This seems to work but I'm a bit sketchy on the details of `Fiber`